core/src/main/java/com/linecorp/armeria/internal/server/websocket/DefaultWebSocketService.java

```diff
@@ -279,18 +284,26 @@ private HttpResponse checkOrigin(ServiceRequestContext ctx, RequestHeaders heade
                                    "missing the origin header");
         }
 
-        if (allowedOrigins.isEmpty()) {
-            // Only the same-origin is allowed.
-            if (!isSameOrigin(ctx, headers, origin)) {
+        if (originPredicate != null) {
```

Comment:
Again, I prefer that `originPredicate` and `allowedOrigins` are evaluated with a logical OR. If `originPredicate` is non-null, the `allowedOrigins` will be ignored

Reply:
I've changed it to use logical `OR` if both `originPredicate` and `allowedOrigins` are configured.